src/pygeoprocessing/geoprocessing.py

```diff
@@ -345,14 +349,27 @@ def raster_calculator(
     try:
         last_time = time.time()
 
+        block_offset_list = list(iterblocks(
+            (target_raster_path, 1), offset_only=True,
+            largest_block=largest_block))
+
         if calc_raster_stats:
             # if this queue is used to send computed valid blocks of
             # the raster to an incremental statistics calculator worker
             stats_worker_queue = queue.Queue()
             exception_queue = queue.Queue()
+
+            if sys.version_info >= (3, 8):
+                block_size_bytes = (
+                    numpy.dtype(numpy.float64).itemsize *
```

Comment:
That was the original reason, but thinking about it for a second, we already know what the target type is so why not just allocate that much ram only? changed in 1f19d0d4851df3d338709130ab5193695bc5634a

Reply:
Actually the real reason was because it was supposed to be a `float64`! The stats worker accumulates variables in that type since it's high precision, so we gotta make sure the underlying memory can hold it too. I added a comment for that.